### ISSUE TITLE
Deprecate sqlite3.version and sqlite3.version_info

### ISSUE THREAD

**Issue by erlend-aasland** [python/cpython#93370]
The `sqlite3.version` and `sqlite3.version_info` attributes have historically created some confusion, as they are sometimes mistaken for the SQLite 3 version number. They used to reflect the _pysqlite_[^1] version number, until the external pysqlite package stopped upstreaming changes to the stdlib sqlite3 module[^2], but today they carry no meaning or practical value[^3].

In order to avoid future frustration, I suggest to deprecate `sqlite3.version` and `sqlite3.version_info` in 3.12, and remove them in 3.14.

See also [discussion on Discourse](https://discuss.python.org/t/want-to-update-sqlite-that-came-with-python/13056/6?u=erlendaasland)

[^1]: The sqlite3 stdlib module was forked off of the external pysqlite module around 2005.
[^2]: Somewhere around 2015, IIRC.
[^3]: `sqlite3.version` is set to the value `'2.6.0'`, and `sqlite3.version_info` is set to the value `(2, 6, 0)`.

**Comment by serhiy-storchaka** [comment]
But it would be nice to expose the Sqlite 3 version number. What do you think?

**Comment by erlend-aasland** [comment]
> 

We already do :) `sqlite3.sqlite_version` and `sqlite3.sqlite_version_info`

**Comment by subramaniam02** [comment]
Not sure what is the process. But I am interested to work on this.

**Comment by erlend-aasland** [comment]
> Not sure what is the process. But I am interested to work on this.

Sorry for the late reply. Looks like there is already a PR up, however there are probably more deprecations coming up in the 3.12 dev phase :)

**Comment by subramaniam02** [comment]
> > Not sure what is the process. But I am interested to work on this.
> 
> Sorry for the late reply. Looks like there is already a PR up, however there are probably more deprecations coming up in the 3.12 dev phase :)

Ok Thank you !

**Comment by erlend-aasland** [comment]
Resolved with #93482

**Comment by erlend-aasland** [comment]
Reopening as pythoninfo.py was not updated to catch the deprecation warning. Excerpt from https://github.com/python/cpython/runs/7056328858?check_suite_focus=true:

```
/Users/runner/work/cpython/cpython/Lib/test/pythoninfo.py:55: DeprecationWarning: version is deprecated and will be removed in Python 3.14
```

**Comment by erlend-aasland** [comment]
Remaining issue resolved by #94277.

**Comment by erlend-aasland** [comment]
There's a remaining use in `Lib/test/test_sqlite3/__init__.py`
